rosetta-lang/src/main/java/com/regnosys/rosetta/utils/MetaUtil.java

```diff
@@ -0,0 +1,27 @@
+package com.regnosys.rosetta.utils;
+
+import java.util.HashSet;
+import java.util.List;
+
+import com.google.common.collect.Sets;
+import com.regnosys.rosetta.types.RMetaAnnotatedType;
+import com.regnosys.rosetta.types.RMetaAttribute;
+
+public class MetaUtil {
+
+	public static List<RMetaAttribute> intersectMeta(RMetaAnnotatedType t1, RMetaAnnotatedType t2) {
+		return Sets
+				.intersection(new HashSet<>(t1.getMetaAttributes()), new HashSet<>(t2.getMetaAttributes()))
+				.immutableCopy()
+				.asList();
+	}
+	
+	public static boolean hasSupersetOfMetaAttributes(RMetaAnnotatedType t1, RMetaAnnotatedType t2) {
+		if (t1.getMetaAttributes().isEmpty() && t2.getMetaAttributes().isEmpty()) {
+			return true;
+		}
+		HashSet<RMetaAttribute> t1Metas = new HashSet<>(t1.getMetaAttributes());
+		HashSet<RMetaAttribute> t2Metas = new HashSet<>(t2.getMetaAttributes());
+		return t1Metas.containsAll(t2Metas);
+	}
```

Comment:
Is this ever used outside of the subtype relation? If not - might consider making this private in there.

Reply:
moved